reconstruct/reconstruct.go

```diff
@@ -101,9 +100,72 @@ func ReconAccuracy(simFilename string, reconFilename string, leavesOnlyFile stri
 	acc := 100 - accuracy
 	answer[allNodes] = acc
 
+	return answer, accByBase
+}
+
+// ReconAccuracyByBase will run if the option given to ReconAccuracy = 1. This function calculates the percentage of first, second and third bases in codons that were correct.
+func ReconAccuracyByBase(simFilename string, reconFilename string, gpFilename string) map[string][]float64 {
+	sim := fasta.Read(simFilename)
+	rec := fasta.Read(reconFilename)
+	recon := fasta.ToMap(rec)
+	genes := genePred.Read(gpFilename)
+	answer := make(map[string][]float64)
+
+	for s := 0; s < len(sim); s++ {
+		var percentage0, percentage1, percentage2 float64
+		var mistakes0, mistakes1, mistakes2 float64
+		var total0, total1, total2 float64
+		rSeq, ok := recon[sim[s].Name]
+		if ok {
```

Comment:
Could we add an "else" to this "if" that would give an error?  Seems like we should always be able to find the seq in recon, right?